addons/account_payment/wizards/res_config_settings.py

```diff
@@ -1,9 +1,77 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     pay_invoices_online = fields.Boolean(config_parameter='account_payment.enable_portal_payment')
+    onboarding_payment_provider = fields.Selection(
+        selection=[('razorpay', "Razorpay"), ('stripe', "Stripe")],
+        compute="_compute_onboarding_payment_provider",
+        store=True,
+    )
+    providers_state = fields.Selection(
+        selection=[('primary_provider', "Primary Provider"), ('other', "Other Provider")],
+        compute='_compute_providers_state',
+        default=False,
+    )
+
+    # === COMPUTE METHODS === #
+
+    @api.depends('company_id.country_id', 'company_id.currency_id')
+    def _compute_onboarding_payment_provider(self):
+        for config in self:
+            if config.company_id.currency_id.name == 'INR':
+                config.onboarding_payment_provider = 'razorpay'
+            elif config.company_id.country_id.is_stripe_supported_country:
+                config.onboarding_payment_provider = 'stripe'
```

Comment:
```suggestion
            config.onboarding_payment_provider = (
               config.company_id.currency_id.name == 'INR' and 'razorpay'
               or config.company_id.country_id.is_stripe_supported_country and  'stripe'
           )
```
Personal preference but you can keep it as it is